src/js/common/schemaform/Address.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import _ from 'lodash';
+import { set } from 'lodash/fp';
+
+import ErrorableSelect from '../components/form-elements/ErrorableSelect';
+import ErrorableTextInput from '../components/form-elements/ErrorableTextInput';
+import { countries, states } from '../utils/options-for-select';
+
+/**
+ * Input component for an address.
+ *
+ * No validation is provided through a currently stubbed isAddressValid function.
+ */
+class Address extends React.Component {
+  constructor(props) {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+    this.state = { value: props.formData, touched: { street: false, city: false, state: false, country: false, postalCode: false } };
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('address-input-');
+  }
+
+  handleBlur(field) {
+    const newState = _.set(['touched', field], true, this.state);
+    this.setState(newState, () => {
+      if (newState.touched.street && newState.touched.city && newState.touched.state && newState.touched.country && newState.touched.postalCode) {
+        this.props.onBlur(this.props.id, newState.value);
+      }
+    });
+  }
+
+  handleChange(path, update) {
+    let newState = _.set(this.state, ['value', path], update);
+    newState = _.set(newState, ['touched', path], true);
+    
+    // if country is changing we should clear the state
+    if (path === 'country') {
+      newState = _.set(newState, ['value', 'state'], '');
+    }
+
+    this.props.onChange(newState.value);
+  }
+
+  isMilitaryCity(city) {
+    const lowerCity = city.toLowerCase().trim();
+
+    return lowerCity === 'apo' || lowerCity === 'fpo' || lowerCity === 'dpo';
+  }
+
+  render() {
+    let stateList = [];
+    const selectedCountry = this.props.formData.country.value || this.props.formData.country;
```

Comment:
One thing we can do here is set formData to the default value when it's undefined:

```
let formData = this.props.formData;
if (typeof formData === 'undefined') {
  formData = getDefaultFormState(this.props.schema, formData, this.props.schema.definitions);
}
```

Then formData.country should always be populated, because it's defaulted to USA in the address schema.

You'll need to import that `getDefaultFormState` function from the library:

`import { getDefaultFormState } from 'react-jsonschema-form/lib/utils';`